java/src/jmri/jmrix/loconet/slotmon/SlotMonDataModel.java

```diff
@@ -468,10 +470,21 @@ public void setValueAt(Object value, int row, int col) {
                             return;
                         }
                     }
+                    // set speed zero
+                    msg = new LocoNetMessage(4);
+                    msg.setOpCode(LnConstants.OPC_LOCO_SPD);
+                    msg.setElement(1, s.getSlot());
+                    msg.setElement(2, 0);
+                    memo.getLnTrafficController().sendLocoNetMessage(msg);
+                    // Delay here allows command station time to xmit on the rails.
+                    try {
+                        Thread.sleep(100);
```

Comment:
This is asking the Swing event thread to wait, which will cause everything that's happening (display, other system connections, etc) to wait.  That's not a good thing. The LnTrafficController has internal queues and is supposed to be handling things like this itself (and there are sections of it that do throttle, though perhaps not for speed messages).  So I'd greatly prefer that you _not_ have a wait here, and that we build it in to the LnTrafficController as needed.